lib/stdlib/test/json_SUITE.erl

```diff
@@ -0,0 +1,507 @@
+        string => String,
+        null => nil
+    },
+
+    Data = <<"{\"a\": [[], {}, true, false, null, {\"foo\": \"baz\"}], \"b\": [1, 2.0, \"three\"]}">>,
+    {Decoded, Acc, <<>>} = json:decode(Data, ok, Decoders),
+    ?assertEqual(ok, Acc),
+    Expected = #{a => [array, object, true, false, nil, #{foo => baz}], b => [1, 2.0, three]},
+    ?assertEqual(Expected, Decoded),
+    ExpectedHistory =
+        [
+            {object_start, ok, []},
+            {string, <<"a">>, a},
+            {array_start, [], []},
+            {array_push, {array, []}, [array]},
+            {array_push, {object, [array]}, [object, array]},
+            {array_push, {true, [object, array]}, [true, object, array]},
+            {array_push, {false, [true, object, array]}, [false, true, object, array]},
+            {array_push, {nil, [false, true, object, array]}, [nil, false, true, object, array]},
+            {object_start, [nil, false, true, object, array], []},
+            {string, <<"foo">>, foo},
+            {string, <<"baz">>, baz},
+            {object_push, {foo, baz, []}, [{foo, baz}]},
+            {object_finish, [{foo, baz}], #{foo => baz}},
+            {array_push, {#{foo => baz}, [nil, false, true, object, array]}, [
+                #{foo => baz}, nil, false, true, object, array
+            ]},
+            {array_finish, [#{foo => baz}, nil, false, true, object, array], [
+                array, object, true, false, nil, #{foo => baz}
+            ]},
+            {object_push, {a, [array, object, true, false, nil, #{foo => baz}], []}, [
+                {a, [array, object, true, false, nil, #{foo => baz}]}
+            ]},
+            {string, <<"b">>, b},
+            {array_start, [{a, [array, object, true, false, nil, #{foo => baz}]}], []},
+            {integer, <<"1">>, 1},
+            {array_push, {1, []}, [1]},
+            {float, <<"2.0">>, 2.0},
+            {array_push, {2.0, [1]}, [2.0, 1]},
+            {string, <<"three">>, three},
+            {array_push, {three, [2.0, 1]}, [three, 2.0, 1]},
+            {array_finish, [three, 2.0, 1], [1, 2.0, three]},
+            {object_push, {b, [1, 2.0, three], [{a, [array, object, true, false, nil, #{foo => baz}]}]}, [
+                {b, [1, 2.0, three]}, {a, [array, object, true, false, nil, #{foo => baz}]}
+            ]},
+            {object_finish, [{b, [1, 2.0, three]}, {a, [array, object, true, false, nil, #{foo => baz}]}], #{
+                a => [array, object, true, false, nil, #{foo => baz}], b => [1, 2.0, three]
+            }}
+        ],
+    ?assertEqual(ExpectedHistory, lists:reverse(get(history))).
+
+set_history(Ty, Acc, Res) ->
+    History = get(history),
+    Entry = {Ty, Acc, Res},
+    put(history, [Entry | History]),
+    Res.
+
+decode(Bin) -> json:decode(Bin).
+
+%%
+%% JSON SUITE tests
+%%
+
+test_json_test_suite(Config) ->
+    DataDir = proplists:get_value(data_dir, Config),
+    TestFiles = filelib:wildcard("*.json", DataDir),
```

Comment:
Testfiles not included in this PR, should it be, what license?

Reply:
This was added now. The files are MIT-licensed - I added such a note in a README file with the folder with test files, as well as a link to the original source.